src/generator/declaration_visitor.ts

```diff
@@ -1,26 +1,57 @@
-import {FunctionDeclaration, FunctionExpression, isIdentifier, LVal, VariableDeclarator} from '@babel/types';
+import {
+    AssignmentExpression,
+    FunctionDeclaration,
+    isArrayExpression,
+    isIdentifier,
+    LVal,
+    VariableDeclarator,
+} from '@babel/types';
 import Visitor from '../visitor';
-type Mapping = Map<string, number>;
+
+type VariableMapping = Map<string, number>;
 
 class DeclarationVisitor extends Visitor {
 
+    private static memoryOffset: number = 0;
```

Comment:
Make this variable non-static otherwise multiple transpile calls share the same `memoryOffset`

Reply:
We decided to move the memory offset calculation into an own visitor to avoid the necessity for a static variable.